website/addons/base/views.py

```diff
@@ -643,6 +643,10 @@ def addon_view_or_download_file(auth, path, provider, **kwargs):
     if action == 'download':
         return redirect(file_node.generate_waterbutler_url(**dict(extras, direct=None, version=version.identifier)))
 
+    if action == 'get_guid':
```

Comment:
This exposes an endpoint to programmatically generate GUIDs. This was implemented in https://github.com/CenterForOpenScience/osf.io/pull/5836 , but we decided to put that on hold because we didn't want to expose GUID creation quite yet.

This is a potential blocker; let's continue discussion of this on the ticket.